src/main/java/org/opensearch/knn/index/codec/KNN920Codec/KNN920Codec.java

```diff
@@ -49,4 +63,48 @@ public DocValuesFormat docValuesFormat() {
     public CompoundFormat compoundFormat() {
         return knnFormatFacade.compoundFormat();
     }
+
+    @Override
+    public KnnVectorsFormat knnVectorsFormat() {
+        return perFieldKnnVectorsFormat;
+    }
+
+    /**
+     * Class provides per field format implementation for Lucene Knn vector type
+     */
+    class KNN920PerFieldKnnVectorsFormat extends PerFieldKnnVectorsFormat {
+
+        @Override
+        public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
+            if (isKnnVectorFieldType(field)) {
+                final KNNVectorFieldMapper.KNNVectorFieldType type = (KNNVectorFieldMapper.KNNVectorFieldType) mapperService.get()
+                    .fieldType(field);
+                final Map<String, Object> params = type.getKnnMethodContext().getMethodComponent().getParameters();
+                final KnnVectorsFormat luceneHnswVectorsFormat = new Lucene92HnswVectorsFormat(
+                    getMaxConnections(params),
+                    getBeamWidth(params)
+                );
+                return luceneHnswVectorsFormat;
+            }
+            return new Lucene92HnswVectorsFormat();
+        }
+
+        private boolean isKnnVectorFieldType(final String field) {
+            return mapperService.isPresent() && mapperService.get().fieldType(field) instanceof KNNVectorFieldMapper.KNNVectorFieldType;
+        }
+
+        private int getMaxConnections(final Map<String, Object> params) {
```

Comment:
May be flip the if condition for better readability?

Reply:
ack